examples/phone-chatbot/bot_runner.py

```diff
@@ -29,218 +37,544 @@
+        errors.append(
+            "'dialout_settings' can only be used with 'voicemail_detection', not with 'call_transfer'"
+        )
+
+    # If we have any errors, raise an exception with all the problems
+    if errors:
+        error_message = "Invalid configuration: " + "; ".join(errors)
+        raise HTTPException(status_code=400, detail=error_message)
+
+
+def ensure_prompt_config(body: Dict[str, Any]) -> Dict[str, Any]:
+    """Ensures the body has appropriate prompts settings, but doesn't add defaults.
+
+    Only makes sure the prompt section exists, allowing the bot script to handle defaults.
+
+    Args:
+        body: The configuration dictionary
+
+    Returns:
+        Updated configuration with prompt settings section
+    """
+    if "prompts" not in body:
+        body["prompts"] = []
+    return body
+
+
+def create_call_transfer_settings(body: Dict[str, Any]) -> Dict[str, Any]:
+    """Create call transfer settings based on configuration and customer mapping.
+
+    Args:
+        body: The configuration dictionary
+
+    Returns:
+        Call transfer settings dictionary
+    """
+    # Return existing settings if already specified
+    if "call_transfer" in body:
+        return body["call_transfer"]
+
+    # Default transfer settings
+    transfer_settings = {
+        "mode": DEFAULT_MODE,
+        "speakSummary": DEFAULT_SPEAK_SUMMARY,
+        "storeSummary": DEFAULT_STORE_SUMMARY,
+        "testInPrebuilt": DEFAULT_TEST_IN_PREBUILT,
+    }
+
+    # Check if we have dialin settings
+    if "dialin_settings" in body:
+        # Create a temporary routing manager just for customer lookup
+        call_config_manager = CallConfigManager(body)
 
+        # Get caller info
+        caller_info = call_config_manager.get_caller_info()
+        from_number = caller_info.get("caller_number")
 
-async def _create_daily_room(
-    room_url, callId, callDomain=None, dialoutNumber=None, vendor="daily", detect_voicemail=False
-):
+        if from_number:
+            # Get customer name from phone number
+            customer_name = call_config_manager.get_customer_name(from_number)
+
+            # If we know the customer name, add it to the config for the bot to use
+            if customer_name:
+                transfer_settings["customerName"] = customer_name
+
+    return transfer_settings
+
+
+# ----------------- Daily Room Management ----------------- #
+
+
+async def create_daily_room(room_url: Optional[str]):
+    """Create or retrieve a Daily room with appropriate properties.
+
+    Args:
+        room_url: Optional existing room URL
+
+    Returns:
+        Dict containing room URL, token, and SIP endpoint
+    """
     if not room_url:
-        # Create base properties with SIP settings
+        # Create base properties
         properties = DailyRoomProperties(
             sip=DailyRoomSipParams(
                 display_name="dialin-user", video=False, sip_mode="dial-in", num_endpoints=1
```

Comment:
you may need to num_endpoints =2, if we do warm transfers between two daily rooms (more like future proof it).